internal/jwtauth/backend.go

```diff
@@ -94,6 +99,28 @@ func (b *multiroleJWTAuthBackend) getClient(config *multiroleJWTConfig) (*vault.
 		return nil, fmt.Errorf("failed to create vault client: %w", err)
 	}
 
-	b.client = client
-	return client, nil
+	b.policyClient = &vaultClient{
+		Client:    client,
+		mountPath: config.JWTAuthPath,
+	}
+	return b.policyClient, nil
+}
+
+type vaultClient struct {
+	*vault.Client
+	mountPath string
+}
+
+func (c *vaultClient) policies(ctx context.Context, request schema.JwtLoginRequest) ([]string, error) {
+	r, err := c.Client.Auth.JwtLogin(ctx, request, vault.WithMountPath(c.mountPath))
+	if err != nil {
+		return nil, fmt.Errorf("vault error: %w", err)
+	}
+	return r.Auth.Policies, nil
+}
+
+type mockPolicyFetcher func(context.Context, schema.JwtLoginRequest) ([]string, error)
```

Comment:
Why do you want the mock/fake here? Isn't it part of the test? And I would call this a fake, and not a mock.

Reply:
Appears it must be in the same package as the interface it's implementing. And the original PR had a separate package for testing, meaning it had to be in the non-testing package.  Moved and renamed it now :)